core/commands/dag/dag.go

```diff
@@ -53,9 +54,15 @@ type ResolveOutput struct {
 	RemPath string
 }
 
+type CarImportStats struct {
+	BlockCount        uint64
+	PayloadBytesCount uint64
```

Comment:
`Payload` can be confusing here:  is this raw blocks (CAR-metadata) or actual data in each block (CAR-metadata-dagmetadata)?


Perhaps renaming it to `BlockBytesCount` and setting this to `nd.Stat().BlockSize()` is a way to remove confusion while keeping this useful no matter what codecs are used inside of blocks?

Reply:
Renamed `PayloadBytesCount` to `BlockBytesCount`

`nd.Size()` and `nd.Stat().BlockSize` return the same value, so I think it would be better to use `nd.Size()`, given the [comment here](https://github.com/ipfs/go-ipld-format/blob/master/format.go#L40).

Reply:
Hm.. if they are the same, it makes sense, but something feels off when I compare Size imported from CAR with value reported by `ipfs dag stat`: 

```console
$ ipfs dag stat bafybeihcyruaeza7uyjd6ugicbcrqumejf6uf353e5etdkhotqffwtguva                                        
Size: 27676801, NumBlocks: 383

$ ipfs dag export bafybeihcyruaeza7uyjd6ugicbcrqumejf6uf353e5etdkhotqffwtguva > test.car                           
 0s  26.41 MiB / ? [--------------------------------------------------------------------------------=-----------------------] 390.25 MiB/s 0s
 
$ ipfs dag import --stats test.car
Pinned root	bafybeihcyruaeza7uyjd6ugicbcrqumejf6uf353e5etdkhotqffwtguva	success
Imported 383 blocks (125832269 bytes)
```

125832269 bytes is ~125 MB which is way more than 26MB

Reply:
@lidel What I did (using `nd.Size()`) worked for the car files in `sharness/t0054-dag-car-import-export-data/` but does not work for your example.  Using `nd.Stat().DataSize + nd.Stat().LinksSize` works for your dag, but not for the test cars/dags.  For the test cars/dags have the stats for almost all blocks as all zeros.

It appears that `nd.Size()` returns `nd.Stat().CumulativeSize` if a block has stat values.  Othersize, `nd.Size()` is set to `len(nd.RawData())`.  This makes both `nd.Size()` and `nd.Stats()` completely unreliable across different dags.

Apparently, the way to get a reliable size is to always use  `len(nd.RawData())`. So, that is what the latest change does.

Reply:
`DataSize` and `LinksSize` sounds like a dag-pb thing, which probably won't be represented in the fixture files.

What stat are we actually after here, @ribasushi what's your expectation of what this sizing is going to report? I would think that it's the size of the output, which includes CAR header, CID lengths and even varint section size prefixes. But I could find that by measuring the size of the output myself, so the utility doesn't seem great. But what is the utility of reporting just the block sizes? What is the useful for?

Reply:
I would expect the `ipfs dag stat` number. It is useful in terms of "this is the amount of IPLD-data these blocks hold"

Reply:
thanks @ribasushi, `len(nd.RawData())` is probably the one then, but in this case maybe just use `len(block.RawData())` then you get to use the block as it comes out of the CAR rather than whatever happens to it through a `Decode` cycle (probably the same, but seems safer to use the one closer to the original)